Palace/Settings/TPPSettingsAccountsList.swift

```diff
@@ -171,8 +171,15 @@
   }
   
   private func updateList(withAccount account: Account) {
-    userAddedSecondaryAccounts.append(account.uuid)
+    if !userAddedSecondaryAccounts.contains(account.uuid) {
+      userAddedSecondaryAccounts.append(account.uuid)
+    }
     updateSettingsAccountList()
+    // Return from search screen to the list of libraries
+    updateNavBar()
```

Comment:
Shouldn't need this anymore since we are jumping directly to the catalog after adding a new account
```suggestion
```